ULL-IACTEC Integration of USB Infrared & Ultraviolet Cameras

fmarcanoull opened the pull request:
Integration of Thermal Expert Infrared & PCO Ultraviolet USB Cameras. Project developed by University of La Laguna & IAC Medical Technologies. First version. Changes developed by Francisco J. Marcano Serrano (fmarcano@ull.edu.es)

fmarcanoull committed:
Paso 1 integración de Thermal EV2 y PCO Ultraviolet

 Cámaras  Thermal EV2 y PCO Ultraviolet  enviando imágenes via PlusServer.

fmarcanoull committed:
ThermalExpert_PCO_Integration

Integration of Thermal Expert Infrared & PCO Ultraviolet Cameras. First version. This project is a join effort between University of La Laguna & IAC Medical Technologies.

Sunderlandkyl commented:
I don't think that the SDK files should be included in the PlusBuild repository unless there is a specific reason to do so.

FindXYZ.cmake should check the default SDK install location, and use that SDK if found. Is "$ENV{APPDATA}/PCO Digital Camera Toolbox/pco.sdk" the default directory for the PCO SDK?

fmarcanoull committed:
Removing ThirdParty folder

Test folder having SDK libraries. Deleted.

fmarcanoull commented:
Yes it is. "$ENV{APPDATA}/PCO Digital Camera Toolbox/pco.sdk" the default directory for the PCO SDK. 
The ThirdParty folder was a preliminary test that I forgot to remove. It's been deleted.

Sunderlandkyl commented on Modules/FindPCOUVSDK.cmake:
```diff
@@ -0,0 +1,79 @@
+  REQUIRED_VARS PCOUV_SDK_DIR PCOUV_SDK_LIBRARY PCOUV_SDK_INCLUDE_DIR
+  VERSION_VAR PCOUV_SDK_VERSION
+)
+#find_package_handle_standard_args(PCOUVSDK
```
Remove unused code.

fmarcanoull replied:
Code removed.

Sunderlandkyl commented on Modules/FindTEEV2SDK.cmake:
```diff
@@ -0,0 +1,92 @@
+
+  IF (CMAKE_HOST_WIN32 AND NOT CMAKE_CL_64 )
+    SET(PLATFORM_SUFFIX "x86")
+#	SET(BIN_FILENAME "${PLATFORM_SUFFIX}_${FILENAME}")
```
Remove unused code.

fmarcanoull replied:
Code removed.

Sunderlandkyl commented on Modules/FindTEEV2SDK.cmake:
```diff
@@ -0,0 +1,92 @@
+  ENDIF (CMAKE_HOST_WIN32 AND NOT CMAKE_CL_64 )
+
+
+#  find_library(TEEV2_SDK_LIBRARY
```
Remove unused code.

fmarcanoull replied:
Code removed.

Sunderlandkyl commented on Modules/FindTEEV2SDK.cmake:
```diff
@@ -0,0 +1,92 @@
+            PATHS "${TEEV2_SDK_DIR}/lib/${PLATFORM_SUFFIX}/" NO_DEFAULT_PATH
+            PATH_SUFFIXES ${PLATFORM_SUFFIX})
+
+#  find_path(TEEV2_SDK_BINARY
```
Remove unused code.

fmarcanoull replied:
Code removed.

Sunderlandkyl commented on Modules/FindPCOUVSDK.cmake:
```diff
@@ -0,0 +1,79 @@
+#  FOUND_VAR PCOUVSDK_FOUND
+#  REQUIRED_VARS PCOUV_SDK_DIR PCOUV_SDK_BINARY PCOUV_SDK_LIBRARY PCOUV_SDK_INCLUDE_DIR
+#  VERSION_VAR PCOUV_SDK_VERSION
+#)
```
End file with newline. See https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline.

fmarcanoull replied:
Newline added.

Sunderlandkyl commented on Modules/FindTEEV2SDK.cmake:
```diff
@@ -0,0 +1,92 @@
+  FOUND_VAR TEEV2SDK_FOUND
+  REQUIRED_VARS TEEV2_SDK_DIR TEEV2_SDK_BINARY TEEV2_SDK_LIBRARY TEEV2_SDK_INCLUDE_DIR
+  VERSION_VAR TEEV2_SDK_VERSION
+)
```
End file with newline. See https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline.

fmarcanoull replied:
Newline added.

Sunderlandkyl commented on Modules/FindTEEV2SDK.cmake:
```diff
@@ -0,0 +1,92 @@
+#  TEEV2SDK_BINARY_DIR           Binaries needed to use EV2 SDK
+
+SET(TEEV2_SDK_PATH_HINTS
+  "../ThirdParty/TE_SDK"
```
Update list of path hints, since SDK isn't in ThirdParty or Plus-bin.

Is there a path hint for the default TEEV2 SDK install?

fmarcanoull replied:
There's no default path for this SDK. Is there a default path that I should use as hint for this case?

Sunderlandkyl replied:
"../../TE_SDK" would be fine, but you can leave it the way it is. If there is no default path, we would expect he user to define TEEV2_SDK_DIR anyways.

Sunderlandkyl commented on Modules/FindPCOUVSDK.cmake:
```diff
@@ -0,0 +1,79 @@
+#  PCOUVSDK_BINARY_DIR           Binaries needed to use UV SDK
+
+SET(PCOUV_SDK_PATH_HINTS
+  "../ThirdParty/PCO_SDK"
```
Update list of path hints, since SDK isn't in ThirdParty or Plus-bin.

fmarcanoull replied:
List of paths updated.

Sunderlandkyl commented on Modules/FindTEEV2SDK.cmake:
```diff
@@ -0,0 +1,92 @@
+
+  IF (CMAKE_HOST_WIN32 AND CMAKE_CL_64 )
+    SET(PLATFORM_SUFFIX "x64")
+	SET(SDK_FILENAME "i3system_TE_x64.dll")
```
Indentation

fmarcanoull replied:
Indentation fixed.

Sunderlandkyl commented on Modules/FindTEEV2SDK.cmake:
```diff
@@ -0,0 +1,92 @@
+
+  # Libraries
+  SET(PLATFORM_SUFFIX "")
+  SET(BIN_FILENAME "")
```
BIN_FILENAME and FILENAME don't seem to be used anywhere.

fmarcanoull replied:
Unused code removed.

Sunderlandkyl reviewed (commented):
Thanks! There are just a few more comments on this PR.

fmarcanoull committed:
Fixing FindTEEV2SDK.cmake, FindPCOUVSDK.cmake

Several fixes applied, hint list updated, unused code removed, newline at eof added, indentation fixed.  Following review: https://github.com/PlusToolkit/PlusBuild/pull/89#pullrequestreview-1287816092

fmarcanoull committed:
Update FindPCOUVSDK.cmake

fmarcanoull committed:
Merge branch 'ULL_IACTEC'
